core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala

```diff
@@ -224,3 +230,30 @@ class OpWorkflowModelReader(val workflowOpt: Option[OpWorkflow]) extends MLReade
   }
 
 }
+
+private object FileReader {
+
+  def loadFile(pathString: String)(implicit conf: Configuration): String = {
+    val path = new Path(pathString)
+    val fs = path.getFileSystem(conf)
+    val allFiles = fs.listFiles(path, false)
+    var partPath: Option[Path] = None
+    while (allFiles.hasNext) {
+      val p = allFiles.next().getPath
+      if (p.getName.startsWith("part")) {
+        partPath = Option(p)
+      }
+    }
+    val finalPath = partPath.getOrElse(path)
```

Comment:
If this is meant for reading single part files, would we want to drop this `.getOrElse(path)` in favor of an exception?

Reply:
that is there because some of our backwards compatibility tests use non part file names. this is meant to read a single workflow .json file if you use our save method this is a part file if you have passed it around it may have a name